src/game/objects/trapdoor.c

```diff
@@ -3,16 +3,39 @@
 #include "game/control.h"
 #include "global/vars.h"
 
-void SetupTrapDoor(OBJECT_INFO *obj)
+static bool TrapDoor_OnTrapDoor(ITEM_INFO *item, int32_t x, int32_t z);
+
+static bool TrapDoor_OnTrapDoor(ITEM_INFO *item, int32_t x, int32_t z)
```

Comment:
Maybe rename to `TrapDoor_On` or `TrapDoor_StandingOn` or something? IMO we don't need to repeat `TrapDoor` here.

Reply:
Fixed